tv_apps/browser/js/bookmarkStore.js

```diff
@@ -82,6 +119,70 @@
       }
     },
 
+//IFDEF_FIREFOX_SYNC
+    _getRootFolderCache: function () {
+      return Promise.all([
+        new Promise(resolve => {
+          // We query remote bookmarks by parentid with
+          // SYNC_BOOKMARK_ROOT_FOLDER_ID for checking if the indexedDB has
+          // remote bookmark data. If there is any remote data, we create a
+          // folder item to show the synced bookmark.
+          SyncBrowserDB.getBookmark({
+            parentid: SYNC_BOOKMARK_ROOT_FOLDER_ID
+          }, syncPlacesBookmarks => {
+            resolve(syncPlacesBookmarks);
+          });
+        }),
+        new Promise(resolve => {
+          // If there is mobile folder and also has children in it,
+          // create a folder item to show the synced mobile bookmark
+          SyncBrowserDB.getBookmark({
+            parentid: SYNC_BOOKMARK_MOBILE_FOLDER_ID
+          }, syncMobileBookmarks => {
+            resolve(syncMobileBookmarks);
+          });
+        }),
+        new Promise(resolve => {
+          // get bookmark data from origin indexdDB
+          BrowserDB.getBookmarks(localBookmarks => {
+            resolve(localBookmarks);
+          });
+        })
```

Comment:
Based on the [code](https://github.com/mozilla-b2g/gaia/pull/32797/files#diff-ed4ae19263bf74556f7542b37c38dbefR204), it is necessary to keep the execution sequence.
Please use 'then' to handle these three cases.